v2/internal/attacktechniques/aws/persistence/iam-create-backdoor-role/main.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus/mitreattack"
+	"log"
+)
+
+//go:embed malicious_policy.json
+var maliciousIamPolicy string
+
+var roleName string = "malicious-iam-role"
+var adminPolicyArn string = "arn:aws:iam::aws:policy/AdministratorAccess"
+
+func init() {
+	const codeBlock = "```"
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:           "aws.persistence.iam-create-backdoor-role",
+		FriendlyName: "Create a new backdoor IAM Role",
+		Description: `
+Establishes persistence by creating a backdoored new role with a policy allowing it to be assumed from an external AWS account and administrative permissions.
+
+Warm-up: None.
+
+Detonation: 
+
+- Create a new IAM role with an assume role policy backdoored, making it accessible from an external, fictitious AWS account:
+
+` + codeBlock + `
+` + maliciousIamPolicy + `
+` + codeBlock + `
+
+and attach the 'AdministratorAccess' managed IAM policy to it. 
+
+References:
+
+- https://www.invictus-ir.com/news/the-curious-case-of-dangerdev-protonmail-me
+`,
+		Detection: `
+- Through [IAM Access Analyzer](https://docs.aws.amazon.com/IAM/latest/UserGuide/access-analyzer-resources.html#access-analyzer-iam-role), 
+which generates a finding when a role can be assumed from a new AWS account or publicly.
+
+- Identify a call to <code>CreateRole</code> closely followed by <code>AttachRolePolicy</code> with an administrator policy.
+
+- Identify a call to <code>CreateRole</code> that contains an assumeRolePolicyDocument in the requestParameters that allows access from an external AWS account. Sample event:
+
+` + codeBlock + `
+{
+  "eventSource": "iam.amazonaws.com",
+  "eventName": "CreateRole",
+  "requestParameters": {
+	"roleName": "malicious-iam-role",
+	"assumeRolePolicyDocument": "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Effect\": \"Allow\",\n      \"Principal\": {\n        \"Service\": \"ec2.amazonaws.com\"\n      },\n      \"Action\": \"sts:AssumeRole\"\n    },\n    {\n      \"Effect\": \"Allow\",\n      \"Principal\": {\n        \"AWS\": \"arn:aws:iam::193672423079:root\"\n      },\n      \"Action\": \"sts:AssumeRole\"\n    }\n  ]\n}"
```

Comment:
love that you included a sample event! same question for the `ec2` thingy as below